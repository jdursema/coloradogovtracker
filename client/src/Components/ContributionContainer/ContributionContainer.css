.contribution-card-container {
  display: flex;
  flex-direction: row;
  margin-left: auto;
  margin-right: auto;
  width: 750px; }

.contribution-card-holder {
  display: flex;
  flex-wrap: wrap;
  }

.active {
  background: green;
  color: white;
}

@media screen and (max-width: 1045px) {
  .contribution-card-container {
    width: 500px; }
}


@media screen and (max-width: 840px) {
  .contribution-card-container {
    width: 250px; } }