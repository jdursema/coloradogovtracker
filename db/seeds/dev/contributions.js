const contributionsData = require('../../../data/contributionsSeed.js');
const candidatesData = require('../../../data/candidateLookup');

exports.seed = function(knex, Promise) {
  // Deletes ALL existing entries
  return knex('contributors').del()
    .then(() => knex('candidates').del())
    .then(() => {
      // Inserts seed entries
      let candidatePromises = [];
      let contributionPromises = [];
      candidatesData.forEach((candidate) => {
        candidatePromises.push(createCandidate(knex, candidate));
      });
      contributionsData.forEach((contribution) => {
        contributionPromises.push(createContributor(knex, contribution));
      });
      return Promise.all([...candidatePromises, ...contributionPromises]);
    });
};



const createCandidate = (knex, candidate) => {
  return knex('candidates').insert(candidate);
};

const createContributor = (knex, contributor) => {
  return knex('contributors').insert(contributor);
};
