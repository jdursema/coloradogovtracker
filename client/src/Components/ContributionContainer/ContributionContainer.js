import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter} from 'react-router-dom';
import * as actions from '../../Actions/';
import { getSelectedCandidate } from '../../Helper/helper';
import Card from '../Card/Card.js'
import './ContributionContainer.css'



class ContributionContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentlyDisplayed: []
    }
  }


componentWillMount() {
  if(this.props.contributions){
     const sortedContributions = this.props.contributions.sort((a, b) => {
      return b.contribution_amount - a.contribution_amount;
    });
    this.setState({currentlyDisplayed:this.props.contributions})
  }
}

componentWillReceiveProps(nextProps) {
  if(this.props != nextProps) {
 
     const sortedContributions = nextProps.contributions.sort((a, b) => {
      return b.contribution_amount - a.contribution_amount;
    });
    this.setState({currentlyDisplayed: nextProps.contributions})
  }
}

  searchContributors = (event) => {
    let contributions = this.props.contributions;
    let searchValue = event.toUpperCase();
    let contributionFilter = contributions.filter(contribution => contribution.donor_last.toUpperCase().includes(searchValue))
  
    this.setState({currentlyDisplayed: contributionFilter})
  
  }

    sortHighContributions = () => {
    const sortedContributions = this.state.currentlyDisplayed.sort((a, b) => {
      return b.contribution_amount - a.contribution_amount;
    });
    this.setState({currentlyDisplayed: sortedContributions});
  }

  sortLowContributions = () => {
     const sortedContributions = this.state.currentlyDisplayed.sort((a, b) => {
      return a.contribution_amount - b.contribution_amount;
    });
    this.setState({currentlyDisplayed: sortedContributions});
  }

  alphabetizContributors = () => {

    const alphabetizedContributors = this.state.currentlyDisplayed.sort((a, b) => {
      if(a.donor_last.split(' ')[0] === null) return 0;
      if (a.donor_last.split(' ')[0] < b.donor_last.split(' ')[0]) return -1;
      if (a.donor_last.split(' ')[0] > b.donor_last.split(' ')[0]) return 1;
      return 0;
    });

    this.setState({currentlyDisplayed: alphabetizedContributors});
    
  }

    deAlphabetizContributors = () => {
      const alphabetizedContributors = this.state.currentlyDisplayed.sort((a, b) => {
      
        if (a.donor_last.split(' ')[0] > b.donor_last.split(' ')[0]) return -1;
        if (a.donor_last.split(' ')[0] < b.donor_last.split(' ')[0]) return 1;
          return 0;
      });

      this.setState({currentlyDisplayed: alphabetizedContributors});
    }


  mapContributions = (contributions, index) => {
    if(contributions) {
      

      const contributionMap = contributions.map((contribution, index) => {
        return (
          <Card 
            id = {contribution.id}
            amount = {contribution.contribution_amount}
            firstName = {contribution.donor_first}
            lastName = {contribution.donor_last}
            amount = {contribution.contribution_amount}
            recordId = {contribution.record_id}
            date = {contribution.contribution_date}
            type = {contribution.contribution_type}
            occupation = {contribution.donor_occupation}
            />
        )
      })
        return contributionMap;
    }
  }

  render () {
    return (
      <div className = "contribution-container">
        <h2 className = "center"> Contributions </h2>
        <div className = "candidate-sort-bar center">
          <button className= "filter-button" onClick = {this.sortHighContributions}>Highest Contributions</button>
          <button className= "filter-button" onClick = {this.sortLowContributions}>Lowest Contributions</button>
          <button className= "filter-button" onClick = {this.alphabetizContributors}> Donors A to Z </button>
          <button className= "filter-button" onClick = {this.deAlphabetizContributors}> Donors Z to A </button> 
           
            <input 
              className = "filter-search"
              onChange = {event => this.searchContributors(event.target.value)}
              type = "text"
              placeholder = "Search Contributors" />
          
        </div>
         <h3 class="center"> Click on the card to see more donor details </h3> 
        <div className = 'contribution-card-container'>

          <div className = 'contribution-card-holder'>
            {this.mapContributions(this.state.currentlyDisplayed)}
          </div> 
        </div>
      </div>
    )
  }
}

export default (ContributionContainer)