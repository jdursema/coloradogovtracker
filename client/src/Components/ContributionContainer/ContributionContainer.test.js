import ContributionContainer from './ContributionContainer'
import React from 'react';
import {shallow} from 'enzyme';
import contributions from '../../mockdata/mockStore.js';

describe('Contribution container tests', () => {
  let mockContributions = contributions;
  let renderedComponent;
  

  beforeEach(() => {
     renderedComponent = shallow(
    <ContributionContainer/>)
  })

it('should render', () => {
  expect(renderedComponent).toBeDefined();
})

it('should match the snapshot', () => {
  expect(renderedComponent).toMatchSnapshot();
})

})