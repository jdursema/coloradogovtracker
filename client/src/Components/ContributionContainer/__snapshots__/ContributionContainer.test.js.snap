// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contribution container tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContributionContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="center"
>
           Contributions 
</h2>,
        <div
          className="candidate-sort-bar center"
>
          <button
                    className="filter-button active"
                    onClick={[Function]}
          >
                    Highest Contributions
          </button>
          <button
                    className="filter-button"
                    onClick={[Function]}
          >
                    Lowest Contributions
          </button>
          <button
                    className="filter-button"
                    onClick={[Function]}
          >
                     Donors A to Z 
          </button>
          <button
                    className="filter-button"
                    onClick={[Function]}
          >
                     Donors Z to A 
          </button>
          <input
                    className="filter-search"
                    onChange={[Function]}
                    placeholder="Search Contributors"
                    type="text"
          />
</div>,
        <h3
          className="center instructions"
>
           Click on the card to see donor details 
</h3>,
        <div
          className="contribution-card-container"
>
          <div
                    className="contribution-card-holder"
          >
                    
          </div>
</div>,
      ],
      "className": "contribution-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Contributions ",
          "className": "center",
        },
        "ref": null,
        "rendered": " Contributions ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="filter-button active"
              onClick={[Function]}
>
              Highest Contributions
</button>,
            <button
              className="filter-button"
              onClick={[Function]}
>
              Lowest Contributions
</button>,
            <button
              className="filter-button"
              onClick={[Function]}
>
               Donors A to Z 
</button>,
            <button
              className="filter-button"
              onClick={[Function]}
>
               Donors Z to A 
</button>,
            <input
              className="filter-search"
              onChange={[Function]}
              placeholder="Search Contributors"
              type="text"
/>,
          ],
          "className": "candidate-sort-bar center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Highest Contributions",
              "className": "filter-button active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Highest Contributions",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lowest Contributions",
              "className": "filter-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Lowest Contributions",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Donors A to Z ",
              "className": "filter-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " Donors A to Z ",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Donors Z to A ",
              "className": "filter-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " Donors Z to A ",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "filter-search",
              "onChange": [Function],
              "placeholder": "Search Contributors",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Click on the card to see donor details ",
          "className": "center instructions",
        },
        "ref": null,
        "rendered": " Click on the card to see donor details ",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="contribution-card-holder"
>
            
</div>,
          "className": "contribution-card-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "contribution-card-holder",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="center"
>
             Contributions 
</h2>,
          <div
            className="candidate-sort-bar center"
>
            <button
                        className="filter-button active"
                        onClick={[Function]}
            >
                        Highest Contributions
            </button>
            <button
                        className="filter-button"
                        onClick={[Function]}
            >
                        Lowest Contributions
            </button>
            <button
                        className="filter-button"
                        onClick={[Function]}
            >
                         Donors A to Z 
            </button>
            <button
                        className="filter-button"
                        onClick={[Function]}
            >
                         Donors Z to A 
            </button>
            <input
                        className="filter-search"
                        onChange={[Function]}
                        placeholder="Search Contributors"
                        type="text"
            />
</div>,
          <h3
            className="center instructions"
>
             Click on the card to see donor details 
</h3>,
          <div
            className="contribution-card-container"
>
            <div
                        className="contribution-card-holder"
            >
                        
            </div>
</div>,
        ],
        "className": "contribution-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Contributions ",
            "className": "center",
          },
          "ref": null,
          "rendered": " Contributions ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="filter-button active"
                onClick={[Function]}
>
                Highest Contributions
</button>,
              <button
                className="filter-button"
                onClick={[Function]}
>
                Lowest Contributions
</button>,
              <button
                className="filter-button"
                onClick={[Function]}
>
                 Donors A to Z 
</button>,
              <button
                className="filter-button"
                onClick={[Function]}
>
                 Donors Z to A 
</button>,
              <input
                className="filter-search"
                onChange={[Function]}
                placeholder="Search Contributors"
                type="text"
/>,
            ],
            "className": "candidate-sort-bar center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Highest Contributions",
                "className": "filter-button active",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Highest Contributions",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lowest Contributions",
                "className": "filter-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Lowest Contributions",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Donors A to Z ",
                "className": "filter-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": " Donors A to Z ",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Donors Z to A ",
                "className": "filter-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": " Donors Z to A ",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "filter-search",
                "onChange": [Function],
                "placeholder": "Search Contributors",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Click on the card to see donor details ",
            "className": "center instructions",
          },
          "ref": null,
          "rendered": " Click on the card to see donor details ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="contribution-card-holder"
>
              
</div>,
            "className": "contribution-card-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "contribution-card-holder",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
